### ISSUE TITLE
[GLPI 9.5.2] - On Locked ticket we can remove approvals!

### ISSUE THREAD

**Issue by andrepetinga** [glpi-project/glpi#8163]
Hello.

**Describe the bug**

We configured the ReadOnly profile with the Create active, otherwise we won't be able to see this when the ticket is locked.

![image](https://user-images.githubusercontent.com/43515938/97008644-ff460f00-153a-11eb-9703-413e655bf5ef.png)

The problem is: There is no "Read" permission for the Validations, only Create. So, in order to make this kind of object to appear with the ReadOnly profile, we had to activate the Create option. But, despite of not activating Purge option, we can Purge Validations with ReadOnly profile!!!

![image](https://user-images.githubusercontent.com/43515938/97008598-f2292000-153a-11eb-953e-848cc6c7088d.png)

Can you please check this?
Thanks.

**Comment by andrepetinga** [comment]
Just tested it on 9.5.6 and still a Bug! I still can delete with Read Only active.

**Comment by tomolimo** [comment]
Hello @andrepetinga 
Could you check the rights on "Validations" for the read-only profile? And could you post a screen-copy?
Thank you
Regards,
Tomolimo

**Comment by andrepetinga** [comment]
Hello @tomolimo .
Still the same rights:

![image](https://user-images.githubusercontent.com/43515938/134035169-da4cf8ee-5e6c-4ce8-9f22-0463ca6cd78f.png)

**Comment by andrepetinga** [comment]
Still happens.

**Comment by blippercop** [comment]
*push*

**Comment by cconard96** [comment]
Tested on latest 9.5 commit.

1. Created new user with read-only profile only
2. Opened ticket with an approval
3. Did not see the approval at all
By default, to read-only profiles shouldn't have any access to create approvals.

I added to permission to create approvals and could then see the approval and edit it but still couldn't delete it.

In the past some of the default profiles had extra permissions accidentally set that may not have shown in the UI properly. Can you check the rights value for the read-only profile in the DB?

Assuming your read-only profile has the ID 8:
```mysql
SELECT rights FROM `glpi_profilerights` WHERE `profiles_id` = '8' AND `name` = 'ticketvalidation';
```
For me with the create right set for requests and incidents, it is 3072.

**Comment by andrepetinga** [comment]
Tested today on 9.5.6 (with no extra updates).

Yes, running that query I got 3072.
Yes, creating a new profile with read-only profile only, works.

But that is not what we want. We have our technicians with their own profile, and when they open a ticket, by default that ticket is read-only (meaning their active profile is now read-only). And despite this, they can Edit and Delete an approval.

**Comment by cconard96** [comment]
I made a new user with the read-only profile to make it easier to test and because it *should* be the exact same experience as a regular user looking at a locked Ticket.

I tested again with locks and it clearly isn't the same experience. I see the same issue now.

I need to see how locks are being handled because it doesn't seem to even change the active profile...

**Comment by cedric-anne** [comment]
Fixed by #10399.
